src/hotspot/os_cpu/linux_riscv/os_linux_riscv.cpp

```diff
@@ -350,6 +351,74 @@ void os::print_context(outputStream *st, const void *context) {
     st->print_cr("%-*.*s=" INTPTR_FORMAT, 8, 8, reg_abi_names[r], (uintptr_t)uc->uc_mcontext.__gregs[r]);
   }
   st->cr();
+  const struct __riscv_mc_d_ext_state * const f_ext_state = &(uc->uc_mcontext.__fpregs.__d);
+  st->print_cr("Floating point state:");
+  st->print_cr("fcsr=" UINT32_FORMAT, f_ext_state->__fcsr);
+  st->print_cr("Floating point registers:");
+  for (int r = 0; r < 32; r++) {
+    st->print_cr("f%d=" INTPTR_FORMAT " | %g", r, (intptr_t)f_ext_state->__f[r], (double)f_ext_state->__f[r]);
+  }
+  st->cr();
+
+#ifdef NO_RVV_SIGCONTEXT
+  st->print_cr("Vector state: JVM compiled without vector sigcontext support");
+#else // ifndef NO_RVV_SIGCONTEXT
+// This magic number is not in any user-space header.
+// No other choice but to define it.
+#ifndef RISCV_V_MAGIC
+#define RISCV_V_MAGIC 0x53465457
+#endif
+
+  // vector state
+  struct __riscv_extra_ext_header *ext;
+  struct __riscv_v_ext_state *v_ext_state;
+
+  // Find the vector context
+  ext = (struct __riscv_extra_ext_header *)(&uc->uc_mcontext.__fpregs);
+  if (ext->hdr.magic != RISCV_V_MAGIC) {
+    st->print_cr("Vector state: not found");
+    return;
+  }
+
+  // The size passed to user-space is calculated accordingly:
+  // size = sizeof(struct __riscv_ctx_hdr) + sizeof(struct __sc_riscv_v_state) + riscv_v_vsize;
+  uint32_t ext_size = ext->hdr.size;
+
+  if (ext_size < (sizeof(struct __riscv_ctx_hdr) + sizeof(*v_ext_state))) {
+    st->print_cr("Vector state: not found, invalid size");
+    return;
+  }
+
+  ext_size -= (sizeof(struct __riscv_ctx_hdr) + sizeof(*v_ext_state));
+
+  v_ext_state = (struct __riscv_v_ext_state *)((char *)(ext) + sizeof(*ext));
+
+  st->print_cr("Vector state:");
+  st->print_cr("vstart=" INTPTR_FORMAT, v_ext_state->vstart);
+  st->print_cr("vl=" INTPTR_FORMAT, v_ext_state->vl);
+  st->print_cr("vtype=" INTPTR_FORMAT, v_ext_state->vtype);
+  st->print_cr("vcsr=" INTPTR_FORMAT, v_ext_state->vcsr);
+  st->print_cr("vlenb=" INTPTR_FORMAT, v_ext_state->vlenb);
```

Comment:
Maybe add some formatting here so that the numbers printed are on the same colume.